web/client/components/widgets/enhancers/emptyChartState.js

```diff
@@ -1,11 +1,13 @@
 import React from 'react';
 import Message from '../../I18N/Message';
 import emptyState from '../../misc/enhancers/emptyState';
+import EmptyView from '../../misc/EmptyView';
 
 export default emptyState(
     ({data = []}) => !data || data.length === 0,
     ({mapSync, iconFit} = {}) => ({
         iconFit,
         tooltip: mapSync ? <Message msgId="widgets.errors.nodatainviewport" /> : <Message msgId="widgets.errors.nodata" />
-    })
+    }),
+    () => <EmptyView content={<Message msgId="widgets.errors.nodatainviewport" />}/>
```

Comment:
in this case we are adding a custom component for the empty state and in this way we are overriding the ternary that select a different message based on the mapSync props.
I think we should keep the two different messages with something like this:
```js
import React from 'react';
import { Glyphicon } from 'react-bootstrap';
import Message from '../../I18N/Message';
import emptyState from '../../misc/enhancers/emptyState';

function ChartEmptyState({ message }) {
    return (
        <div
            style={{
                position: 'absolute',
                width: '100%',
                height: '100%',
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center'
            }}>
            <Glyphicon glyph="stats"/>&nbsp;{message}
        </div>
    );
}

export default emptyState(
    ({data = []}) => !data || data.length === 0,
    ({mapSync} = {}) => ({
        message: mapSync ? <Message msgId="widgets.errors.nodatainviewport" /> : <Message msgId="widgets.errors.nodata" />
    }),
    ChartEmptyState
);
```

Note: the above code is only an example to discuss and on how we could implement this.